harness/determined/profiler.py

```diff
@@ -704,11 +758,87 @@ def run(self) -> None:
             )
             batch_start_time = cast(float, batch_start_time)
             if time.time() - batch_start_time > self.FLUSH_INTERVAL:
-                self.send_queue.put(self.current_batch.convert_to_post_format())
-                self.current_batch.clear()
+                self.send_queue.put(self.metrics_batch.consume())
                 batch_start_time = time.time()
 
 
+class MetricBatch:
+    def __init__(self, trial_id: str, agent_id: str) -> None:
+        self.trial_id = trial_id
+        self.agent_id = agent_id
+        self.batch = {}  # type: Dict[Tuple[MetricType, str, str], List[Measurement]]
+
+    def append(
```

Comment:
nit: I personally wouldn't call this `append` because `append` is a threadsafe operation in Python while this operation is not. Not very important, so feel free to leave it as `append` if you think that's the best name.